pyfr/plugins/tavg.py

```diff
@@ -357,3 +362,248 @@ def __call__(self, intg):
                     self.tstart_acc = intg.tcurr
 
                 self.tout_last = intg.tcurr
+
+
+class TavgCLIPlugin(BaseCLIPlugin):
+    name = 'tavg'
+
+    @classmethod
+    def add_cli(cls, parser):
+        cmd = cls.name
```

Comment:
Maybe have this bit done by the calling code and just pass in sp.

Reply:
It seems to me best to have this done here in the object that defines the cli plugin.

Reply:
The lines are going to be generic for all CLI plugins: get our name, add a parser, add a sub parser.  I'd rather plugins have no choice about their command name (same way plugins don't get to decide their section names).

Reply:
However, every CLI plugin might have different requirements in terms of inputs and outputs. Previously I had a method where the plugin defined an attribute list that was used to automatically define the parser for the cli. But you asked for the current method instead.

Reply:
It is always going to be `parser.add_parser(cmd, help=f'{cmd} --help')`.  What the plugin does beyond this though is it's call (in terms of subcommands, flags, etc).  But we can certainly factor out the parser creation.